Dockerfile

```diff
@@ -1,10 +1,28 @@
+FROM golang:latest AS build
+
+RUN mkdir -p /go/src/github.com/nsqio/nsq
+COPY . /go/src/github.com/nsqio/nsq
+
+WORKDIR /go/src/github.com/nsqio/nsq
+
+RUN go get github.com/mattn/goveralls \
+ && wget -O /bin/dep https://github.com/golang/dep/releases/download/v0.3.2/dep-linux-amd64 \
+ && chmod +x /bin/dep \
+ && /bin/dep ensure \
+ && ./test.sh \
+ && ./coverage.sh --coveralls \
```

Comment:
I don't think we need to run code coverage as part of this build?

Reply:
@mreiferson Why not? In my projects I always have testing/coveraging as part of build process. For example if one test fails - build will be interrupted and broken image will not be uploaded to Docker repository. I think this is good practice. Yes, it has some impact on build time, but it is not too drastical.

Reply:
Runnings tests here can be useful because they can fail the build. But in this case the coverage results are just thrown out.